### DISCUSSION TITLE
Network partition

### DISCUSSION THREAD

**Discussion by qazwsxedckll** [olric-data/olric#246] [Q&A] [answered]
> When the network healed, the stopped nodes joins again the cluster and fragmented partitions is merged by their primary owners in accordance with LWW policy.

I have done some experiments on hashicorp/memberlist. If I isolate some node from others, they will form a new cluster and never rejoin when the network is healed. It seams that `func (d *Discovery) Join() (int, error)` is only called once on start. I want to know how olric can handle this situation.

**Comment by buraksezer** [comment, marked as answer]
Olric implements simple split-brain protection system. I don't remember the details of how memberlist works in that manner but you can check the documentation and the code: https://github.com/buraksezer/olric?tab=readme-ov-file#simple-split-brain-protection
